spec/features/ingredients/index_spec.rb

```diff
@@ -0,0 +1,26 @@
+require "rails_helper"
+
+RSpec.describe "the ingredients index" do
+  before(:each) do
+    @ingredient_1 = Ingredient.create!(name: "Ground Beef", cost: 2)
+    @ingredient_2 = Ingredient.create!(name: "Salt", cost: 4)
+    @ingredient_3 = Ingredient.create!(name: "Pepper", cost: 4)
+    @ingredient_4 = Ingredient.create!(name: "Avocado", cost: 4)
+  end
+
+  it "lists all the ingredients attributes" do
+    visit "/ingredients"
+    
+    expect(page).to have_content(@ingredient_1.name)
+    expect(page).to have_content(@ingredient_1.cost)
+    expect(page).to have_content(@ingredient_2.name)
+    expect(page).to have_content(@ingredient_2.cost)
+  end
+  
+  it "sorts the ingredients alphabetically" do
+    visit "/ingredients"
+
+    expect(Ingredient.sort_alphabetically).to eq([@ingredient_4, @ingredient_1, @ingredient_3, @ingredient_2])
```

Comment:
Since this is a feature test, we want to test from the perspective of the user. In this case, we can use the `orderly` gem to do something like:
`expect(@ingredient_4).to appear_before(@ingredient_1)`
`expect(@ingredient_1).to appear_before(@ingredient_3)`
`expect(@ingredient_3).to appear_before(@ingredient_2)`

The feature tests typically shouldn't call a model method, but rather just make assertions about what is rendered on the page after performing user actions (i.e. clicking buttons, filling out forms, visiting pages, etc)

Reply:
The style of test you have here would be better suited for a model test in the ingredient_spec.rb